pkg/provider/config/azure.go

```diff
@@ -168,6 +169,19 @@ type Config struct {
 	ClusterServiceSharedLoadBalancerHealthProbePort int32 `json:"clusterServiceSharedLoadBalancerHealthProbePort,omitempty" yaml:"clusterServiceSharedLoadBalancerHealthProbePort,omitempty"`
 	// ClusterServiceSharedLoadBalancerHealthProbePath defines the target path of the shared health probe. Default to `/healthz`.
 	ClusterServiceSharedLoadBalancerHealthProbePath string `json:"clusterServiceSharedLoadBalancerHealthProbePath,omitempty" yaml:"clusterServiceSharedLoadBalancerHealthProbePath,omitempty"`
+
+	// RetrievedClusterPodCidr tracks whether the pod subnet prefix for the cluster has been retrieved and set within the Network Security Group (NSG).
+	// The NSG for the pod subnet is configured once for the entire cluster.
+	// If the pod subnet prefix is not set, the value is false.
+	// If the pod subnet prefix is set, the value is true.
+	// Note: Multiple rules per subnet with different protocols and destination ports can exist.
+	RetrievedClusterPodCidr bool `json:"retrievedClusterPodCidr" yaml:"retrievedClusterPodCidr"`
+
+	// PodCidrIPv4 is the IPv4 pod subnet prefix for the cluster.
+	// PodCidrIPv6 is the IPv6 pod subnet prefix for the cluster.
+	// The pod subnet prefix is used to configure the NSG for the pod subnet.
```

Comment:
Could you also add a message to say all Pod CIDR would be opened to internet by default?

and add a TODO comment to improve the security later?